src/main/java/seedu/address/model/person/Birthday.java

```diff
@@ -0,0 +1,48 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.time.LocalDate;
+
+/**
+ * Represents a Person's birthday in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidBirthday(String)}
+ */
+public class Birthday {
+
+    public static final String MESSAGE_CONSTRAINTS = "Birthdays should come in the form of yyyy-MM-dd";
+    public static final String VALIDATION_REGEX = "^\\d{4}\\-(0?[1-9]|1[012])\\-(0?[1-9]|[12][0-9]|3[01])$";
+    public final LocalDate birthday;
+
+    /**
+     * Constructs an {@code Birthday}.
+     *
+     * @param birthday A valid date.
+     */
+    public Birthday(String birthday) {
+        requireNonNull(birthday);
+        checkArgument(isValidBirthday(birthday), MESSAGE_CONSTRAINTS);
+        this.birthday = LocalDate.parse(birthday);
```

Comment:
Might this lead to parsing error for dates that fit the Regex but is not an actual date? Eg. 31 Feb

Reply:
It might be helpful to have a differentiating message between when user format is wrong and when user enters a invalid date.